pkg/scheduler/api/unschedule_info.go

```diff
@@ -56,13 +56,24 @@ func (f *FitErrors) SetNodeError(nodeName string, err error) {
 	default:
 		fe = &FitError{
 			NodeName: nodeName,
-			Reasons:  []string{obj.Error()},
+			Status:   []*Status{{Code: Error, Reason: obj.Error()}},
 		}
 	}
 
 	f.nodes[nodeName] = fe
 }
 
+// NotHelpfulForPreemptNodes returns the set of nodes that has no help from preempting pods from it
+func (f *FitErrors) NotHelpfulForPreemptNodes() map[string]struct{} {
+	ret := make(map[string]struct{})
+	for _, node := range f.nodes {
+		if node.Status.ContainsUnschedulableAndUnresolvable() {
+			ret[node.NodeName] = struct{}{}
```

Comment:
we can use `sets.Empty{}` to replace `struct{}`.

Reply:
Mark it, do it in next Pr.

Reply:
Done.